### DISCUSSION TITLE
[V5] CORS error on default API setup

### DISCUSSION THREAD

**Discussion by ghost** [adonisjs/core#2646] [Help] [answered]
By default it is blocking all origins, I have already tried to change the `config/cors.ts` file and even adding the exact origeins does not work.

I was paranoid thinking it was my specific application, but I created an API structure from scratch, just added a POST route and deployed to heroku, the problem remains.

![error](https://i.ibb.co/1mkxTLT/Captura-de-Tela-2021-05-19-a-s-13-23-14.png)

![error](https://i.ibb.co/XXMk0GP/Captura-de-Tela-2021-05-19-a-s-13-23-20.png)

I also switched browsers, it didn't work.

my `config/cors.ts`

```ts
import { CorsConfig } from '@ioc:Adonis/Core/Cors'
const corsConfig: CorsConfig = {
  enabled: false,
  origin: true, // '*' or ['http://localhost:3000'] has no effect
  methods: ['GET', 'HEAD', 'POST', 'PUT', 'PATCH', 'DELETE'],
  headers: true,
  exposeHeaders: [
    'cache-control',
    'content-language',
    'content-type',
    'expires',
    'last-modified',
    'pragma'
  ],
  credentials: true,
  maxAge: 90
}
export default corsConfig
```

I have this problem after the release. 

Thank you all ❤️

**Comment by ghost** [comment, marked as answer]
my stupidity, I didn't pay attention to the `enabled: false` that was previously coming as `true`, resolved!

**Reply by yarapolana** [reply]
Weird I am having the same problem although by default cors enabled is true.

**Reply by thetutlage** [reply]
Can you share your cors config file?

**Reply by yarapolana** [reply]
```js
import { CorsConfig } from '@ioc:Adonis/Core/Cors'

const corsConfig: CorsConfig = {
  enabled: true,
  origin: [
    'https://app.oktaves.com',
  ],
  methods: ['GET', 'HEAD', 'POST', 'PUT', 'DELETE'],
  headers: true,
  exposeHeaders: [
    'cache-control',
    'content-language',
    'content-type',
    'expires',
    'last-modified',
    'pragma',
  ],
  credentials: true,
  maxAge: 90,
}

export default corsConfig

```

**Reply by yarapolana** [reply]
The same happens with socket.io

```js
import { Server } from 'socket.io'
import AdonisServer from '@ioc:Adonis/Core/Server'

class Ws {
  public io: Server
  public booted = false

  public boot() {
    if (this.booted) {
      return
    }

    this.booted = true
    this.io = new Server(AdonisServer.instance!, {
      cors: {
        origin: '*',
      },
    })
  }
}

/* Initiate service to handle sockets */
export default new Ws()

```

**Reply by maswebber** [reply]
> ```js
> import { CorsConfig } from '@ioc:Adonis/Core/Cors'
> 
> const corsConfig: CorsConfig = {
>   enabled: true,
>   origin: [
>     'https://app.oktaves.com',
>   ],
>   methods: ['GET', 'HEAD', 'POST', 'PUT', 'DELETE'],
>   headers: true,
>   exposeHeaders: [
>     'cache-control',
>     'content-language',
>     'content-type',
>     'expires',
>     'last-modified',
>     'pragma',
>   ],
>   credentials: true,
>   maxAge: 90,
> }
> 
> export default corsConfig
> ```

i deployed on heroku with same config, and got CORS too, the API i am targeting is Google Translate API

**Reply by thetutlage** [reply]
Just to make sure I understand everything. Your frontend app is running on domain `https://app.oktaves.com`. It makes an API request to your AdonisJS backend and receives the CORS error when making an AJAX request?

**Reply by yarapolana** [reply]
Yes, I didn’t notice until I turned off my cors plugin.
Is there something I missed?

**Reply by thetutlage** [reply]
Can you change `origin: true` and see if that works?

**Reply by yarapolana** [reply]
@thetutlage got lost in time, forgot to reply.

`origin: true` and `origin: '*'` **work** but allows some unwanted constant bot visitors

`origin: ['https://app.oktaves.com', ...]` and `origin: ['app.oktaves.com', ...]` **doesn't work** it gives me cors errors

is there a way to add ip address? like this?
Just trying things out.
`origin: <ip_address>`

**Reply by yarapolana** [reply]
@thetutlage
Last update:  I changed all my cors configurations socket, adonis, media server and other configs to `['https://app.oktaves.com', ...]` and now works perfectly. =)

**Comment by yarapolana** [comment]
Yes, I didn’t notice until I turned of my cors plugin.
Is there something I missed?

**Reply by crytos** [reply]
How did you turn off the cors plugin

**Reply by crytos** [reply]
/**
 * Config source: https://git.io/JfefC
 *
 * Feel free to let us know via PR, if you find something broken in this config
 * file.
 */

import { CorsConfig } from '@ioc:Adonis/Core/Cors'

const corsConfig: CorsConfig = {
  /*
  |--------------------------------------------------------------------------
  | Enabled
  |--------------------------------------------------------------------------
  |
  | A boolean to enable or disable CORS integration from your AdonisJs
  | application.
  |
  | Setting the value to `true` will enable the CORS for all HTTP request. However,
  | you can define a function to enable/disable it on per request basis as well.
  |
  */
  enabled: true,

  // You can also use a function that return true or false.
  // enabled: (request) => request.url().startsWith('/api')

  /*
  |--------------------------------------------------------------------------
  | Origin
  |--------------------------------------------------------------------------
  |
  | Set a list of origins to be allowed for `Access-Control-Allow-Origin`.
  | The value can be one of the following:
  |
  | https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin
  |
  | Boolean (true)    - Allow current request origin.
  | Boolean (false)   - Disallow all.
  | String            - Comma separated list of allowed origins.
  | Array             - An array of allowed origins.
  | String (*)        - A wildcard (*) to allow all request origins.
  | Function          - Receives the current origin string and should return
  |                     one of the above values.
  |
  */
  origin: '*',

  /*
  |--------------------------------------------------------------------------
  | Methods
  |--------------------------------------------------------------------------
  |
  | An array of allowed HTTP methods for CORS. The `Access-Control-Request-Method`
  | is checked against the following list.
  |
  | Following is the list of default methods. Feel free to add more.
  */
  methods: ['GET', 'HEAD', 'POST', 'PUT', 'DELETE'],

  /*
  |--------------------------------------------------------------------------
  | Headers
  |--------------------------------------------------------------------------
  |
  | List of headers to be allowed for `Access-Control-Allow-Headers` header.
  | The value can be one of the following:
  |
  | https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Request-Headers
  |
  | Boolean(true)     - Allow all headers mentioned in `Access-Control-Request-Headers`.
  | Boolean(false)    - Disallow all headers.
  | String            - Comma separated list of allowed headers.
  | Array             - An array of allowed headers.
  | Function          - Receives the current header and should return one of the above values.
  |
  */
  headers: true,

  /*
  |--------------------------------------------------------------------------
  | Expose Headers
  |--------------------------------------------------------------------------
  |
  | A list of headers to be exposed by setting `Access-Control-Expose-Headers`.
  | header. By default following 6 simple response headers are exposed.
  |
  | Cache-Control
  | Content-Language
  | Content-Type
  | Expires
  | Last-Modified
  | Pragma
  |
  | In order to add more headers, simply define them inside the following array.
  |
  | https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers
  |
  */
  exposeHeaders: [
    'cache-control',
    'content-language',
    'content-type',
    'expires',
    'last-modified',
    'pragma',
  ],

  /*
  |--------------------------------------------------------------------------
  | Credentials
  |--------------------------------------------------------------------------
  |
  | Toggle `Access-Control-Allow-Credentials` header. If value is set to `true`,
  | then header will be set, otherwise not.
  |
  | https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials
  |
  */
  credentials: false,

  /*
  |--------------------------------------------------------------------------
  | MaxAge
  |--------------------------------------------------------------------------
  |
  | Define `Access-Control-Max-Age` header in seconds.
  | https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age
  |
  */
  maxAge: 90,
}

export default corsConfig

**Comment by crytos** [comment]
import { CorsConfig } from '@ioc:Adonis/Core/Cors'

const corsConfig: CorsConfig = {

  enabled: true,

  origin: '*',
  methods: ['GET', 'HEAD', 'POST', 'PUT', 'DELETE'],

  headers: true,


  exposeHeaders: [
    'cache-control',
    'content-language',
    'content-type',
    'expires',
    'last-modified',
    'pragma',
  ],

  credentials: false,

  maxAge: 90,
}

export default corsConfig
`
